internal/command/user_v3_test.go

```diff
@@ -1101,3 +1103,2078 @@ func TestCommandSide_DeleteSchemaUser(t *testing.T) {
+								1,
+								json.RawMessage(`{
+						"name": "user"
+					}`),
+							),
+						),
+						eventFromEventPusher(
+							schemauser.NewDeletedEvent(context.Background(),
+								&schemauser.NewAggregate("user1", "org1").Aggregate,
+							),
+						),
+					),
+				),
+				checkPermission: newMockPermissionCheckAllowed(),
+			},
+			args: args{
+				ctx:    context.Background(),
+				userID: "user1",
+			},
+			res: res{
+				err: func(err error) bool {
+					return errors.Is(err, zerrors.ThrowNotFound(nil, "COMMAND-G4LOrnjY7q", "Errors.User.NotFound"))
+				},
+			},
+		},
+		{
+			name: "user not active, precondition error",
+			fields: fields{
+				eventstore: expectEventstore(
+					expectFilter(
+						eventFromEventPusher(
+							schemauser.NewCreatedEvent(context.Background(),
+								&schemauser.NewAggregate("user1", "org1").Aggregate,
+								"schema",
+								1,
+								json.RawMessage(`{
+						"name": "user"
+					}`),
+							),
+						),
+						eventFromEventPusher(
+							schemauser.NewLockedEvent(context.Background(),
+								&schemauser.NewAggregate("user1", "org1").Aggregate,
+							),
+						),
+					),
+				),
+				checkPermission: newMockPermissionCheckAllowed(),
+			},
+			args: args{
+				ctx:    context.Background(),
+				userID: "user1",
+			},
+			res: res{
+				err: func(err error) bool {
+					return errors.Is(err, zerrors.ThrowPreconditionFailed(nil, "TODO", "TODO"))
+				},
+			},
+		},
+		{
+			name: "lock user, ok",
+			fields: fields{
+				eventstore: expectEventstore(
+					expectFilter(
+						eventFromEventPusher(
+							schemauser.NewCreatedEvent(context.Background(),
+								&schemauser.NewAggregate("user1", "org1").Aggregate,
+								"schema",
+								1,
+								json.RawMessage(`{
+						"name": "user"
+					}`),
+							),
+						),
+					),
+					expectPush(
+						schemauser.NewLockedEvent(context.Background(),
+							&schemauser.NewAggregate("user1", "org1").Aggregate,
+						),
+					),
+				),
+				checkPermission: newMockPermissionCheckAllowed(),
+			},
+			args: args{
+				ctx:    context.Background(),
+				userID: "user1",
+			},
+			res: res{
+				want: &domain.ObjectDetails{
+					ResourceOwner: "org1",
+				},
+			},
+		},
+		{
+			name: "lock user, no permission",
+			fields: fields{
+				eventstore: expectEventstore(
+					expectFilter(
+						eventFromEventPusher(
+							schemauser.NewCreatedEvent(context.Background(),
+								&schemauser.NewAggregate("user1", "org1").Aggregate,
+								"schema",
+								1,
+								json.RawMessage(`{
+						"name": "user"
+					}`),
+							),
+						),
+					),
+				),
+				checkPermission: newMockPermissionCheckNotAllowed(),
+			},
+			args: args{
+				ctx:    context.Background(),
+				userID: "user1",
+			},
+			res: res{
+				err: func(err error) bool {
+					return errors.Is(err, zerrors.ThrowPermissionDenied(nil, "AUTHZ-HKJD33", "Errors.PermissionDenied"))
+				},
+			},
+		},
+		{
+			name: "lock user, self",
```

Comment:
Does this test expect that a user can lock himself?

Reply:
Status changes have now a separate permission check, so self unlock is not possible anymore